controller/value_resolver.rst

```diff
@@ -205,6 +205,83 @@ PSR-7 Objects Resolver:
     :class:`Psr\\Http\\Message\\RequestInterface` or :class:`Psr\\Http\\Message\\MessageInterface`.
     It requires installing :doc:`the PSR-7 Bridge </components/psr7>` component.
 
+Managing Value Resolvers
+------------------------
+
+For each argument, every resolver tagged with ``controller.argument_value_resolver``
+will be called until one provides a value. The order in which they are called depends
+on their priority. For example, the ``SessionValueResolver`` (priority 50) will be
+called before the ``DefaultValueResolver`` (priority -100) which allows to write e.g.
+``SessionInterface $session = null`` to get the session if there is one, or ``null``
+if there is none.
+
+But what if you *know* there will be a session? In that case every resolver running
+before ``SessionValueResolver`` is useless. Worse, some of these could actually
+provide a value before ``SessionValueResolver`` has a chance to (don't worry though,
+this won't happen with built-in resolvers). Since Symfony 6.3, this kind of issue
+can be resolved by leveraging the
+:class:`Symfony\\Component\\HttpKernel\\Attribute\\ValueResolver` attribute::
+
+    // src/Controller/SessionController.php
+    namespace App\Controller;
+
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\HttpKernel\Attribute\ValueResolver;
+    use Symfony\Component\HttpKernel\Controller\ArgumentResolver\SessionValueResolver;
+    use Symfony\Component\Routing\Annotation\Route;
+    use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+    class SessionController
+    {
+        #[Route('/')]
+        public function __invoke(
+            #[ValueResolver(SessionValueResolver::class)]
+            SessionInterface $session
+        ): Response
+        {
+            // ...
+        }
+    }
+
+.. versionadded:: 6.3
+
+    The ``ValueResolver`` attribute was introduced in Symfony 6.3.
+
+You can target a resolver by passing its name (more on that later) as ``ValueResolver``'s
+first argument. For convenience, built-in resolvers' name are their FQCN.
+
+By default, a targeted resolver is "pinned" to the argument holding the
+``ValueResolver`` attribute, meaning that only it will be called to provide a value,
+and that it will have to.
+
+In the above example the ``DefaultValueResolver`` would never be called, so adding a
+default value to ``$session`` would be useless. If we need one, then it is fine not
+to use ``ValueResolver``.
+But then, what if we want to prevent an hypothetic ``EagerValueResolver`` to provide a
+value before ``SessionValueResolver``? Time to use ``ValueResolver``'s second argument!
+By passing it to ``true``, you can disable the targeted resolver::
```

Comment:
I don't fully understand this. Why do you want to add a value resolver and then disable it? Is disabling it equivalent to not adding the value resolver attribute?

Reply:
The `ValueResolver` attribute does not “add” a value resolver but “targets” one. By default a targeted value resolver is “pinned” to the argument, meaning only it will be called to provide a value.

If there is no pinned value resolver for an argument, we iterate over a list of value resolvers until one provides a value. Disabling a targeted resolver means said resolver is removed from that list.